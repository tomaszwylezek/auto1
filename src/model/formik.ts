import React from 'react';

export interface FormikField {
  onChange: (e: React.ChangeEvent<any>) => void;
  onBlur: (e: any) => void;
  value: any;
  name: string;
}
