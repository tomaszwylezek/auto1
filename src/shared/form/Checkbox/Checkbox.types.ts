export interface ICheckboxOwnProps {
  label?: string;
}

export interface ICheckboxProps extends ICheckboxOwnProps {}
