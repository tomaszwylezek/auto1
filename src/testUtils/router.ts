import { RouteComponentProps } from 'react-router';
import { UnregisterCallback, Href, Location } from 'history';

export function getMockedRouterProps<P = any>(params?: P, locationData?: Partial<Location>) {
  const location: Location = {
    pathname: '',
    hash: '',
    key: '',
    search: '',
    state: '',
    ...locationData
  };

  const props: RouteComponentProps<P> = {
    match: {
      isExact: true,
      params,
      path: '',
      url: ''
    },
    location,
    history: {
      length: 2,
      action: 'POP',
      location,
      push: jest.fn(),
      replace: () => null,
      go: () => null,
      goBack: jest.fn(),
      goForward: () => null,
      block: () => {
        return null as UnregisterCallback;
      },
      createHref: () => {
        const temp: Href = '';
        return temp as Href;
      },
      listen: () => {
        return null as UnregisterCallback;
      }
    },
    staticContext: {}
  };

  return props;
}
