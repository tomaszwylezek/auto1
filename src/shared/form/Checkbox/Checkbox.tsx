import React from 'react';

// Components
import { Label } from '../Label/Label';

// Models
import { FieldProps } from 'formik';
import { ICheckboxProps } from './Checkbox.types';

export const Checkbox: React.FC<ICheckboxProps & FieldProps> = ({ field, form, label, ...props }): JSX.Element => {
  // { name, value, onChange, onBlur }
  return (
    <div className="custom-control custom-checkbox">
      <input
        type="checkbox"
        {...field}
        {...props}
        checked={field.value}
        id={field.name}
        className={'custom-control-input'}
      />
      {!!label && (
        <Label id={field.name} className="custom-control-label">
          {label}
        </Label>
      )}
    </div>
  );
};
