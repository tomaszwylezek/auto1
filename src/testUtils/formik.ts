import { FormikActions, FormikHandlers, FormikProps } from 'formik';
import { FormikField } from 'model';

export function getMockedFormikConfig<T>(options: Partial<FormikProps<T>> = {}): FormikProps<T> {
  return {
    ...getMockedFormikActions(),
    ...getMockedFormikHandles(),
    enableReinitialize: false,
    isValidating: false,
    isSubmitting: false,
    dirty: false,
    isValid: false,
    submitCount: 0,
    registerField: jest.fn(),
    unregisterField: jest.fn(),
    values: {} as T,
    errors: {} as T,
    initialValues: {} as T,
    touched: {} as T,
    ...options
  };
}

export function getMockedFormikActions<V>(options: Partial<FormikActions<V>> = {}): FormikActions<V> {
  return {
    setStatus: jest.fn(),
    setError: jest.fn(),
    setErrors: jest.fn(),
    setSubmitting: jest.fn(),
    setTouched: jest.fn(),
    setValues: jest.fn(),
    setFieldValue: jest.fn(),
    setFieldError: jest.fn(),
    setFieldTouched: jest.fn(),
    validateForm: jest.fn(),
    validateField: jest.fn(),
    resetForm: jest.fn(),
    submitForm: jest.fn(),
    setFormikState: jest.fn(),
    ...options
  };
}

export function getMockedFormikHandles(options: Partial<FormikHandlers> = {}): FormikHandlers {
  return {
    handleSubmit: jest.fn(),
    handleReset: jest.fn(),
    handleBlur: jest.fn(),
    handleChange: jest.fn(),
    ...options
  };
}

export function getMockedField(options: Partial<FormikField> = {}): FormikField {
  return {
    onChange: jest.fn(),
    onBlur: jest.fn(),
    value: '',
    name: '',
    ...options
  };
}
