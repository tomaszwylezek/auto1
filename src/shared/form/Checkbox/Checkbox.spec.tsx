import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { Checkbox } from './Checkbox';

// Models
import { ICheckboxProps } from './Checkbox.types';
import { getMockedField, getMockedFormikConfig } from 'testUtils/formik';
import { FieldProps } from 'formik';

describe('Checkbox component should', (): void => {
  let props: ICheckboxProps & FieldProps;
  let component: ShallowWrapper<ICheckboxProps & FieldProps>;

  beforeEach((): void => {
    props = {
      label: 'bar',
      field: getMockedField({ name: 'foo' }),
      form: getMockedFormikConfig()
    };

    component = shallow(<Checkbox {...props} />);
  });

  it('renders correctly', (): void => {
    expect(component).toMatchSnapshot();
  });

  it('renders invalid message', (): void => {
    component.setProps({
      form: getMockedFormikConfig({
        touched: { foo: true },
        errors: { foo: 'core.error' }
      })
    });

    expect(component).toMatchSnapshot();
  });
});
