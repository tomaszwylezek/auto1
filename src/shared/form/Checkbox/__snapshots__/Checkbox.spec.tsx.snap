// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component should renders correctly 1`] = `
<div
  className="custom-control custom-checkbox"
>
  <input
    checked=""
    className="custom-control-input"
    id="foo"
    name="foo"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    type="checkbox"
    value=""
  />
  <Label
    className="custom-control-label"
    id="foo"
  >
    <FormattedMessage
      id="bar"
    />
  </Label>
</div>
`;

exports[`Checkbox component should renders invalid message 1`] = `
<div
  className="custom-control custom-checkbox"
>
  <input
    checked=""
    className="custom-control-input is-invalid"
    id="foo"
    name="foo"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    type="checkbox"
    value=""
  />
  <Label
    className="custom-control-label"
    id="foo"
  >
    <FormattedMessage
      id="bar"
    />
  </Label>
  <div
    className="invalid-feedback"
  >
    <FormattedMessage
      id="core.error"
    />
  </div>
</div>
`;
